### PR TITLE
Replace license/copyright header for no-go files

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**How to categorize this PR?**

/area compliance
/kind cleanup

**What this PR does / why we need it**:
Replace copyright/license headers for all "non go" files with SPDX format.

### CODE REVIEW

File: .ci/get_latest_release_component_descriptor [hunk 1]
```diff
@@ -1,18 +1,8 @@
 #!/usr/bin/env python3
 #
-# Copyright 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
```
Comment (acumino) [inline .ci/get_latest_release_component_descriptor:3]: I find it weird to change the year to 2024 in every file.
Can you write some script or something so we can preserve the old year?

Comment (RaphaelVogel) [reply .ci/get_latest_release_component_descriptor:3]: Sure could do, but before I make that changes let's wait for some other opinions on that. I already changed around 90 other gardener repos using the current year. In general the year is not really relevant. You can even completely omit the year (https://reuse.software/faq/#years-copyright).
So if other maintainers think the same I will change it.

Comment (ialidzhikov) [reply .ci/get_latest_release_component_descriptor:3]: I would also vote to drop the year if there aren't any consequences/benefits of maintaining it. 
+1 for dropping the year.

Comment (RaphaelVogel) [reply .ci/get_latest_release_component_descriptor:3]: Will check that officially, if removing the year completely has any negative/legal consequences.

Comment (RaphaelVogel) [reply .ci/get_latest_release_component_descriptor:3]: @acumino @ialidzhikov 
According to legal, we still need to add the current year as copyright year. So please continue with reviewing this PR.

Comment (acumino) [reply .ci/get_latest_release_component_descriptor:3]: > According to legal, we still need to add the current year as copyright year. So please continue with reviewing this PR.

In that case, I would prefer to preserve the old year.

Comment (RaphaelVogel) [reply .ci/get_latest_release_component_descriptor:3]: It is not a matter of preference or taste here. It is a strong recommendation from our legal department to include the current year if changes are done to a file. So I think we should follow this recommendation.

Comment (acumino) [reply .ci/get_latest_release_component_descriptor:3]: In the past, we never changed the year when a change was made to the file. 
> It is a strong recommendation from our legal department to include the current year if changes are done to a file.

In that case. I am fine with it.
/lgtm

**Review by rfranzke — COMMENTED** [review]
/approve

**Pull request merged**